docs/video-player.md

```diff
@@ -0,0 +1,128 @@
+
+          # if provided, title is displayed in the upper left corner of the video
+          title
+
+          # if provided, width and height are used to define the aspect ratio of the
+          # player, so to avoid layout jumps during the rendering.
+          width
+          height
+
+          # if provided, it shows a blurred placeholder for the video
+          blurUpThumb
+
+          # you can include more data here: they will be ignored by the component
+        }
+      }
+    }
+  }
+`;
+
+export default withQuery(query)(Page);
+```
+
+## Props
+
+The `<VideoPlayer />` components supports all the properties made available for
+`<MuxPlayer />` component from `@mux/mux-player-react` package plus `data`, which is meant to
+receive data directly in the shape they are provided by DatoCMS GraphQL API.
+
+`<Video Player />` uses the `data` prop to generate a set of props for the inner
+`<MuxPlayer />`. On this topic, also see the "Advanced usage" section below.
+
+| prop | type           | required           | description                                                      | default |
+| ---- | -------------- | ------------------ | ---------------------------------------------------------------- | ------- |
+| data | `Video` object | :white_check_mark: | The actual response you get from a DatoCMS `video` GraphQL query |         |
+
+Compared to the `<MuxPlayer />`, some prop default values are different on `<VideoPlayer />`
+
+- `disableCookies` is normally true, unless you explicitly set the prop to `false`
+- the video height and width, when available in the `data` props, are used to set `{ aspectRatio: "[width] / [height]"}` for the `<MuxPlayer />`'s `style`
+
+All the other props are forwarded to the `<MuxPlayer />` component that is used internally.
+
+## Advanced usage: the `useVideoPlayer` hook
+
+Despite we try our best to make the `<VideoPlayer />` suitable and easy to use for most normal use cases, there are situation where you may need to leverage the `<MuxPlayer />` directly. If that's the case, fill free to use the hook we provide: `useVideoPlayer`.
+
+`useVideoPlayer` takes data coming in the shape they are produced from DatoCMS API and return props that you can pass to dhe `<MuxPlayer />` component: that's pretty much what the `<VideoPlayer />` does internally.
+
+### Example
+
+```
+import 
+
+const data = {
+  muxPlaybackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
+  title: 'Title',
+  width: 1080,
+  height: 1920,
+  blurUpThumb:
+    'data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A',
+};
+
+// `props` is the following object:
+//
+//     {
+//        playbackId: 'ip028MAXF026dU900bKiyNDttjonw7A1dFY',
+//        title: 'Title',
+//        style: {
+//          aspectRatio: '1080 / 1920',
+//        },
+//        placeholder:
+//          'data:image/bmp;base64,Qk0eAAAAAAAAABoAAAAMAAAAAQABAAEAGAAAAP8A',
+//      }
+const props = useVideoPlayer({ data });
```

Comment:
Hmm, if I'm reading this right, does `useVideoPlayer` just convert the shape of `data`(from our graphql) into the shape of `props` (for the Mux component)?

e.g. `props.playbackId = data.muxPlaybackId`, etc.?

If so, it kinda seems like a utility function to me. I wonder if maybe we could just export a pure function, e.g.:

` import {getMuxPropsFromGraphQLData} from @datocms/react-datocms`?

I'm not totally sure about this, but I think it might kinda be an non-standard/unnecessary use of custom react hooks...? I think normally people write hooks to interact with React components, their states/lifecycles, etc. In this case we are just taking one data  object and changing the shape of it and returning another data object, so the hook seems a little overkill, maybe?

(To be clear, this isn't a code quality concern, just a DX concern. Like a "useVideoPlayer" hook makes it sound like some fancy thing that interacts with other components' state, etc. It suggests more power than we actually provide with that hook.)

Reply:
My take on this point is: that is not a proper hook, but:

- its role is quite similar to a hook: it generates stuff used later on to populate the JSX side of the component
- if in the future we want to add some behaviour involving the component state, we will already have the right name in place

Maybe I'm over-engineering?

Reply:
I don't know the right answer to this, sorry :) Your call, or maybe ask the other devs?

Either way, it's a useful function and I'm glad we're providing it!